src/GitHubApps/src/DarcBot/AzureFunction.cs

```diff
@@ -0,0 +1,212 @@
+                    }
+                }
+            }
+
+            _log.LogInformation($"buildId: {triageItem.BuildId}, recordId: {triageItem.RecordId}, index: {triageItem.Index}");
+            _log.LogInformation($"category: {triageItem.UpdatedCategory}");
+            _log.LogInformation($"url: {triageItem.Url}");
+
+            await IngestTriageItemsIntoKusto(new[] { triageItem });
+
+            await gitHubClient.Issue.Comment.Create(repositoryId, prNumber, $"DarcBot has updated the 'TimelineIssuesTriage' database.\n**PowerBI reports may take up to 24 hours to refresh**\n\nSee {docLink} for more information and 'darcbot' usage.");
+            return new OkObjectResult("Success");
+        }
+
+        private static async Task IngestTriageItemsIntoKusto(TriageItem[] triageItems)
+        {
+            _log.LogInformation("Entering IngestTriageItemIntoKusto");
+            string kustoIngestConnectionString = System.Environment.GetEnvironmentVariable("KustoIngestConnectionString");
+            string databaseName = System.Environment.GetEnvironmentVariable("KustoDatabaseName");
+            IKustoIngestClient ingestClient = KustoIngestFactory.CreateQueuedIngestClient(kustoIngestConnectionString);
+            await KustoHelpers.WriteDataToKustoInMemoryAsync(
+                ingestClient,
+                databaseName,
+                "TimelineIssuesTriage",
+                _log,
+                triageItems,
+                b => new[]
+                {
+                    new KustoValue("ModifiedDateTime", b.ModifiedDateTime.ToString(), KustoDataTypes.DateTime),
+                    new KustoValue("BuildId", b.BuildId.ToString(), KustoDataTypes.Int),
+                    new KustoValue("RecordId", b.RecordId.ToString(), KustoDataTypes.Guid),
+                    new KustoValue("Index", b.Index.ToString(), KustoDataTypes.Int),
+                    new KustoValue("UpdatedCategory", b?.UpdatedCategory, KustoDataTypes.String),
+                    new KustoValue("Url", b?.Url, KustoDataTypes.String)
+                });
+        }
+
+        private static string GetDarcBotProperty(string propertyName, string body)
+        {
+            MatchCollection propertyMatches = _darcBotPropertyRegex.Matches(body);
+            foreach (Match propertyMatch in propertyMatches)
+            {
+                if (propertyMatch.Success)
+                {
+                    string key = propertyMatch.Groups["key"].Value;
+                    string value = propertyMatch.Groups["value"].Value;
+                    if (propertyName.Equals(key, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return value;
+                    }
+                }
+            }
+            return null;
+        }
+
+        private static TriageItem GetTriageItemProperties(string body)
+        {
+            TriageItem triageItem = new TriageItem();
+
+            if (!string.IsNullOrEmpty(body))
```

Comment:
You could invert these to reduce nesting:
```csharp
if (string.IsNullOrEmpty(body))
    return null;
```

Reply:
good point